src/Program.ts

```diff
@@ -303,20 +328,29 @@ export class Program {
 
                 //add file to the `source` dependency list
                 if (brsFile.pkgPath.startsWith(startOfSourcePkgPath)) {
+                    this.createSourceScope();
                     this.dependencyGraph.addDependency('scope:source', brsFile.dependencyGraphKey);
                 }
 
                 //add the file to the program
                 this.files[pathAbsolute] = brsFile;
-                let fileContents = await getFileContents();
+                let fileContents: SourceObj = {
+                    pathAbsolute: pathAbsolute,
+                    source: await getFileContents()
+                };
+                this.plugins.emit('beforeFileParse', fileContents);
+
                 this.logger.time(LogLevel.info, ['parse', chalk.green(pathAbsolute)], () => {
-                    brsFile.parse(fileContents);
+                    brsFile.parse(
+                        fileContents.source,
+                        () => this.plugins.emit('afterFileParse', brsFile)
```

Comment:
`BrsFile` already has a reference to `Program`, so I don't think we need to pass in a callback function for parse, because the `BrsFile` can just call `this.program.plugins.emit('afterFileParse', this)`.

Reply:
Good point - that's not needed anymore.